client/src/containers/Admin/Users/AssignUserRole.js

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from "react";
+import {
+  getUserRoles,
+  getRoles,
+  addRoleToUser,
+  clearRolesByUser
+} from "../../../helpers/api";
+
+class AssignUserRole extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userRoles: [],
+      allRoles: [],
+      user_id: "",
+      name: "",
+      email: "",
+      message: "",
+      messageAlert: ""
+    };
+  }
+
+  componentDidMount = async () => {
+    const user_id = this.props.match.params.userId;
+    const userRolesData = await getUserRoles(user_id);
+    const allRolesData = await getRoles();
+    const allRoles = allRolesData.data;
+    const userRoles = userRolesData.data.map(role => role.role);
+    const { name, email } = userRolesData.data[0];
+    this.setState({ user_id, userRoles, allRoles, name, email });
+  };
+
+  handleCheckbox = e => {
+    const value = e.target.value;
+    const isChecked = e.target.checked;
+    if (isChecked) {
+      this.setState({
+        userRoles: [...this.state.userRoles, value]
+      });
+    } else {
+      const userRoles = this.state.userRoles.filter(role => role !== value);
+      this.setState({
+        userRoles
+      });
+    }
+  };
+
+  onSave = async e => {
+    e.preventDefault();
+    const { user_id, allRoles, userRoles } = this.state;
+
+    const roles = allRoles.filter(role => {
+      return userRoles.includes(role.name);
+    });
+
+    if (userRoles.length === 0) {
+      this.setState({
+        message: "You have to assign at least one role to a user.",
+        messageAlert: "alert alert-danger"
+      });
+    } else {
+      try {
+        let res = "";
+        await clearRolesByUser(user_id);
```

Comment:
I think you don't need a specific API to remove the role. Try to do everything in the same API. The problem if you do like this is if cleaerRolesByUser succeeds and for some reason the below addRoleToUser fails, we will end up with a user without any role in the database. I think this should be a kind of atomic action. You just need one API but in that API you can first clear the roles and then add the roles.